tests/parsers/test.installrdf.js

```diff
@@ -22,45 +28,125 @@ describe('InstallRdfParser.getMetaData()', function() {
   it('should collect an error on invalid type value', () => {
     var addonValidator = new Validator({_: ['bar']});
     var rdf = validRDF('<em:type>whatevs</em:type>');
-    var installRdfParser = new InstallRdfParser(rdf, addonValidator.collector);
-    return installRdfParser.getMetaData()
+    var rdfScanner = new RDFScanner(rdf, INSTALL_RDF);
+    return rdfScanner.getContents()
+      .then((xmlDoc) => {
+        var installRdfParser = new InstallRdfParser(xmlDoc,
+                                                    addonValidator.collector);
+        return installRdfParser._getAddonType();
+      })
       .then(() => {
         var errors = addonValidator.collector.errors;
         assert.equal(errors.length, 1);
-        assert.equal(errors[0].code, messages.TYPE_INVALID.code);
+        assert.equal(errors[0].code, messages.RDF_TYPE_INVALID.code);
       });
   });
 
   it('should resolve with mapped type value', () => {
     var rdf = validRDF('<em:type>2</em:type>');
-    var installRdfParser = new InstallRdfParser(rdf);
-    return installRdfParser.getMetaData()
-      .then((addonMetaData) => {
+    var rdfScanner = new RDFScanner(rdf, INSTALL_RDF);
+    return rdfScanner.getContents()
+      .then((xmlDoc) => {
+        var installRdfParser = new InstallRdfParser(xmlDoc);
+        return installRdfParser._getAddonType();
+      })
+      .then((type) => {
         // Type 2 maps to 1 PACKAGE_EXTENSION
-        assert.equal(addonMetaData.type, constants.PACKAGE_EXTENSION);
+        assert.equal(type, constants.PACKAGE_EXTENSION);
       });
   });
 
   it('should resolve with mapped type value for experiments', () => {
     var rdf = validRDF('<em:type>128</em:type>');
-    var installRdfParser = new InstallRdfParser(rdf);
-    return installRdfParser.getMetaData()
-      .then((addonMetaData) => {
+    var rdfScanner = new RDFScanner(rdf, INSTALL_RDF);
+    return rdfScanner.getContents()
+      .then((xmlDoc) => {
+        var installRdfParser = new InstallRdfParser(xmlDoc);
+        return installRdfParser._getAddonType();
+      })
+      .then((type) => {
         // Type 128 (experiments) maps to 1 PACKAGE_EXTENSION
-        assert.equal(addonMetaData.type, constants.PACKAGE_EXTENSION);
+        assert.equal(type, constants.PACKAGE_EXTENSION);
       });
   });
 
   it('should collect a notice if type is missing', () => {
     var addonValidator = new Validator({_: ['bar']});
-    var installRdfParser = new InstallRdfParser(validRDF(''),
-                                                addonValidator.collector);
-    return installRdfParser.getMetaData()
+    var rdfScanner = new RDFScanner(validRDF(''), INSTALL_RDF);
```

Comment:
`validRDF()` should be fine, you don't need the empty string argument. (Though it's not really a `validRDF` anymore, is it :wink: )